### PR TITLE
[MAINT, MRG] Alphabetize ``docdict`` to limit redundancy in the long term

### LINKED ISSUE #10380: [DOC] Alphabetizing `docdict`
I think the original organization of `docdict` was great when there were limited entries but now, in my opinion, it's getting a bit unruly and disorganized. My suggestion would be to alphabetize and document the standardized format that the argument should go first with any specific separated by and underscore, i.e. there can be `trans` and `trans_not_none` which will be next to each other alphabetically and that way it will be easy to tell which docstring (i.e. (`%(color_matplotlib)`) goes to which argument (i.e. `color`) where `%(matplotlib_color)` for the argument color is a bit confusing that it goes to the argument `color` for  a function such as `f(..., color=None, ...)`.

cc @hoechenberger @agramfort

### PR DESCRIPTION
**Pull request by alexrockhill**
Fixes #10380.

### CODE REVIEW

**Comment by alexrockhill** [comment]
The way this was done was to take the existing `docdict` and use it to write to a text file in alphabetized order and with the argument first if it was switched (i.e. `raw_tmin` -> `tmin_raw`) so I think it's pretty safe none of the descriptions got modified besides what I went in and put back how it was with `_argument_base` to remove redundancy.

**Comment by alexrockhill** [comment]
I just realized now `docdict` is a real-world dictionary

**Comment by alexrockhill** [comment]
I'm building the documentation on `main` to compare to this branch now. There should be no difference in the html files, that's how you know it's a good PR ;)

**Comment by alexrockhill** [comment]
Ok no diffs, there were some bits that don't render like this:

```
< L1094-L1113"><span 
---
> L1093-L1112"><span 
```

If anyone wants to check here's some code:

```
diff -qr _build ~/Downloads/_build -x *.pickle -x *.zip -x *.doctree >> tmp.txt
```

```
with open('tmp.txt', 'r') as fid:
     txt = fid.readlines()
with open('tmp.txt', 'w') as fid:
    for line in txt:
        fid.write(line.split(' ')[1] + '\n')
```

```
for FNAME in $(cat tmp.txt); do echo $FNAME; diff <(fold -s -w 20 $FNAME) <(fold -s -w 20 ~/Downloads/$FNAME); done
```

Good to merge by me.

**Comment by agramfort** [comment]
Sometimes I see a - and not a _ in the key of the docdict. Not sure why. Besides LGTM

Thx @alexrockhill

**Comment by alexrockhill** [comment]
> Sometimes I see a - and not a _ in the key of the docdict. Not sure why. Besides LGTM

Those were there before..

**Comment by alexrockhill** [comment]
The failure is a notebook timeout, I don't think it's related

**Comment by larsoner** [comment]
Usually when we implement a standard like this, we try to codify and test it somewhere (this also follows TDD!). Otherwise over time we rely on reviewers to catch deviations, and we inevitably won't...

Would it be reasonable to adapt your script to check that the `docdict` entries are alphabetized? Or, it seems like a few lines based on `re.findall` run on the contents of `docs.py` might do it. If it doesn't seem doable or easy, okay if I take a few minutes and push a commit if I can get something working?

**Comment by alexrockhill** [comment]
You're welcome to!

File: mne/utils/docs.py [hunk 1]
```diff
@@ -261,814 +175,589 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
...
-docdict["topomap_vmin_vmax"] = """
-vmin, vmax : float | callable | None
-    Lower and upper bounds of the colormap, in the same units as the data.
-    If ``vmin`` and ``vmax`` are both ``None``, they are set at ± the
-    maximum absolute value of the data (yielding a colormap with midpoint
-    at 0). If only one of ``vmin``, ``vmax`` is ``None``, will use
-    ``min(data)`` or ``max(data)``, respectively. If callable, should
-    accept a :class:`NumPy array <numpy.ndarray>` of data and return a
-    float.
+docdict['average_plot_psd'] = """
```
Comment (larsoner) [inline mne/utils/docs.py:213]: My test appears to have been useful:
```
E   AssertionError: assert ['accept', 'a...clust_1', ...] == ['accept', 'a...clust_1', ...]
E     At index 20 diff: 'average_plot_psd' != 'average_psd'
E     Use -v to get the full diff
```
Should I push and you can iterate to fix these?

**Comment by alexrockhill** [comment]
Not sure why the notebooks are failing to run but I think this is good to go

**Comment by larsoner** [comment]
Thanks @alexrockhill

**Comment by drammock** [comment]
Just a note to say that this has made some tasks harder for me, not easier. It caters to the use case of *working on a single param entry*, but IMO it's really useful to have adjacent docdict entries be grouped semantically (e.g., all the docdict entries related to PSDs or TFRs are together) rather than lexically (all the entries that start with "P" are together). I can still get the situation I want (e.g., I can do `psd_exclude`, `psd_method`, `psd_db`, etc) but it requires violating the convention that the docdict entry should start with the param name.

I agree that *something* needed to be done, and at least now we have a convention for underscores vs hyphens (though here again, we landed on the wrong convention IMO --- I prefer hyphens because they're fewer keystrokes on my keyboard).

There is also at least one place where a multiline line-end comment got split up because the entries were no longer adjacent, though one could argue that line-end comments that span multiple lines are bad practice anyway.

@alexrockhill WDYT about my use case? Are you (or others) finding the alphabetized organization easier to work with?

**Comment by larsoner** [comment]
> it requires violating the convention that the docdict entry should start with the param name.

Indeed this is what I started doing more of. We already had some of this with `maxwell_filter_*` names I think.

>  prefer hyphens because they're fewer keystrokes on my keyboard

Fine by me to switch gradually or all at once

> Are you (or others) finding the alphabetized organization easier to work with?

At least for me it has been faster to new params because I don't have to think about semantically where it goes, I first just decide the most sane name, and then put it in the one place it can possibly go. Before alphabetizing, I'd search for a bit, maybe find zero, one, or two possible reasonable places it could go, and then have to decide which one it should land in (or if we need  a new category). And then maybe I'd notice an existing entry that I thought would need to move...

In this sense the semantic grouping is more burdensome, because even defining a good set of clear, non-overlapping categories that people should use in the first place will be a tough task / open to interpretation. Then these have to be discoverable/listed somewhere ideally, and then you have to look through a very long file to find all the possible options, etc. Some of these issues can be mitigated of course, but I think it would result in a lot more thought and maintenance work than "alphabetical".

**Comment by alexrockhill** [comment]
I agree that in principle grouping semantically is preferable but in practice it was a mess and led to a lot of duplicate entries. If there was a way to systematically group semantically, that would be preferred but I couldn't think of one that would be easy to implement.

I don't think the gain of organization is so much in adding to the docdict but instead in finding things within the docdict to use so that you don't need to make so many additional entries and can use and find ones that exist already.

**Comment by drammock** [comment]
ok, I'll try working with it for a while and see if I get used to it or if it still chafes. Sounds like @larsoner is OK with occasionally using a semantic prefix (like `maxwell_filter`); but I'll at least try to avoid that if possible and see how it goes.

**Pull request merged**